R/tmb.R

```diff
@@ -145,11 +146,28 @@ h_mmrm_tmb_data <- function(formula_parts,
       formula_parts$full_formula,
       data = data,
       weights = .(as.symbol(weights_name)),
-      na.action = "na.pass"
+      na.action = "na.pass",
+      xlev = xlev
     ))
   )
   if (drop_levels) {
-    full_frame <- droplevels(full_frame, except = formula_parts$visit_var)
+    ix <- vapply(
```

Comment:
this part here gets quite complex, can you factor this out as a helper function with unit tests?

Reply:
now it is using `droplevels` again, but added a helper to check if a factor should drop levels